applications/nrf5340_audio/src/modules/sd_card_playback.c

```diff
@@ -0,0 +1,362 @@
+	uint8_t pcm_wav_mono_frame[pcm_wav_mono_frame_size];
+	ret = sd_card_open(playback_file_name, &f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("Error when trying to open file on SD card. Return value: %d", ret);
+		return ret;
+	}
+
+	sd_card_playback_active = true;
+	while (sd_card_playback_active) {
+
+		ret = sd_card_read(pcm_wav_mono_frame, &pcm_wav_mono_frame_size, &f_seg_read_entry);
+		if (ret < 0) {
+			LOG_ERR("Error when trying to read file on SD card. Return value: %d", ret);
+			return ret;
+		}
+		if (pcm_wav_mono_frame_size < pcm_frame_size) {
+			sd_card_playback_active = false;
+		}
+		/* Wait until there is enough space in the ringbuffer */
+		if (sd_card_playback_active) {
+			k_sem_take(&m_sem_load_from_buf_lc3, K_FOREVER);
+			sd_card_playback_buffer_to_ringbuffer(pcm_wav_mono_frame,
+							      pcm_wav_mono_frame_size);
+		}
+	}
+	ret = sd_card_close(&f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("Error when closing file. Return value: %d", ret);
+		return ret;
+	}
+	sd_card_playback_active = false;
+	return 0;
+}
+
+static void sd_card_playback_thread(void *arg1, void *arg2, void *arg3)
+{
+	while (!sw_codec_is_initialized()) {
+		k_msleep(100);
+	}
+	while (1) {
+		k_sem_take(&m_sem_playback, K_FOREVER);
+		switch (playback_file_format) {
+		case WAV:
+			sd_card_playback_thread_play_wav();
+			break;
+		case LC3:
+			sd_card_playback_thread_play_lc3();
+			break;
+		}
+	}
+}
+
+bool sd_card_playback_is_active(void)
+{
+	return sd_card_playback_active;
+}
+
+void sd_card_playback_wav(char *filename, uint32_t frame_duration_ms, uint8_t bit_depth,
+			  enum sd_playback_sample_rates sample_rate)
+{
+	playback_file_format = WAV;
+	playback_file_name = filename;
+	wav_playback_cfg.frame_duration_ms = frame_duration_ms;
+	wav_playback_cfg.bit_depth = bit_depth;
+	wav_playback_cfg.sample_rate = sample_rate;
+	k_sem_give(&m_sem_playback);
+}
+
+void sd_card_playback_lc3(char *filename)
+{
+	playback_file_format = LC3;
+	playback_file_name = filename;
+	k_sem_give(&m_sem_playback);
+}
+
+int sd_card_playback_mix_with_stream(void *const pcm_a, size_t pcm_a_size)
+{
+	int ret;
+	uint8_t pcm_b[pcm_frame_size];
+	size_t write_size = pcm_frame_size;
+
+	ret = sd_card_playback_buffer_set(pcm_b, &write_size);
+	if (ret < 0) {
+		LOG_ERR("Error when loading pcm data into buffer. Ret: %d", ret);
+		return ret;
+	}
+	if (write_size == pcm_frame_size) {
+		pcm_mix(pcm_a, pcm_a_size, pcm_b, pcm_frame_size, B_MONO_INTO_A_STEREO_L);
+	} else {
+		LOG_WRN("Size written to buffer does not equal the requested size");
+	}
+	return 0;
+}
+
+int sd_card_playback_init(void)
+{
+	int ret;
+
+	sd_card_playback_thread_id = k_thread_create(
+		&sd_card_playback_thread_data, sd_card_playback_thread_stack,
+		SD_CARD_PLAYBACK_STACK_SIZE, (k_thread_entry_t)sd_card_playback_thread, NULL, NULL,
+		NULL, K_PRIO_PREEMPT(4), 0, K_NO_WAIT);
+	ret = k_thread_name_set(sd_card_playback_thread_id, "sd_card_playback");
+	if (ret < 0) {
+		LOG_ERR("Failed");
```

Comment:
No need for a print here. `Failed` does not give the developer much additional information.

Reply:
So I shouldn't print the error code?